codec/src/main/java/io/netty/handler/codec/protobuf/ProtobufVarint32FrameDecoder.java

```diff
@@ -47,31 +48,78 @@
     @Override
     protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) throws Exception {
         in.markReaderIndex();
-        final byte[] buf = new byte[5];
-        for (int i = 0; i < buf.length; i ++) {
-            if (!in.isReadable()) {
-                in.resetReaderIndex();
-                return;
-            }
+        int preIndex = in.readerIndex();
+        int length = readRawVarint32(in);
+        if (preIndex == in.readerIndex()) {
+            return;
+        }
+        if (length < 0) {
+            throw new CorruptedFrameException("negative length: " + length);
+        }
 
-            buf[i] = in.readByte();
-            if (buf[i] >= 0) {
-                int length = CodedInputStream.newInstance(buf, 0, i + 1).readRawVarint32();
-                if (length < 0) {
-                    throw new CorruptedFrameException("negative length: " + length);
-                }
+        if (in.readableBytes() < length) {
+            in.resetReaderIndex();
+            return;
+        } else {
+            out.add(in.readBytes(length));
+            return;
+        }
+    }
 
-                if (in.readableBytes() < length) {
-                    in.resetReaderIndex();
-                    return;
+    /**
+     * Reads variable length 32bit int from buffer
+     *
+     * @return decoded int if buffers readerIndex has been forwarded else nonsense value
+     * @throws IOException
+     */
+    private static int readRawVarint32(ByteBuf buffer) throws IOException {
```

Comment:
Please adjust NOTICE file and add license to licenses folder for the code you forked.